core/os/android/adb/device.go

```diff
@@ -183,6 +205,27 @@ func newDevice(ctx context.Context, serial string, status bind.Status) (*binding
 		d.To.Configuration.PerfettoCapability = perfettoCapability
 	}
 
+	// If the VkRenderStagesProducer layer exist, we assume the render stages producer is
+	// implemented in the layer.
+	for _, l := range d.To.Configuration.GetDrivers().GetVulkan().GetLayers() {
+		capability := d.To.Configuration.PerfettoCapability
+		if capability == nil {
+			capability = &device.PerfettoCapability{
+				GpuProfiling: &device.GPUProfiling{},
+			}
+		}
+		gpu := capability.GpuProfiling
+		if gpu == nil {
+			gpu = &device.GPUProfiling{}
+			capability.GpuProfiling = gpu
+		}
```

Comment:
This should be in the if below, you are doing it for every layer.

Reply:
Done